module/zfs/dmu_send.c

```diff
@@ -1684,9 +1686,15 @@ issue_data_read(struct send_reader_thread_arg *srta, struct send_range *range)
 	    .zb_blkid = range->start_blkid,
 	};
 
+	/*
+	 * vfs_ratelimit_data_read_spin() will sleep in short periods and return
+	 * immediately when a signal is pending.
+	 */
+	vfs_ratelimit_data_read_spin(os, 0, BP_GET_LSIZE(bp));
```

Comment:
Unrelated to this particular chunk, I wonder about expected practice of backup/restore datasets with rate limits.  From the server admin standpoint I am not sure I'd like my backup and especially restore procedures to be throttled by user-oriented limits, or even accounted towards them.

Reply:
Yes, I wasn't sure about that, but users can also do send/recv and those limits are not user-oriented by dataset-oriented, just like quotas vs user-quotas - root cannot go beyond the former. At the very least I could make it a tunable.

Reply:
While you are right that quotas might also cause problems during replication, it is more of an exception, for example if destination uses different compression, pool topology, etc.  Here though it will be a daily issue, and I don't see a good solution, since there might be user and admin workload going same time, and disabling rates for the first to help second is not perfect either.  It would have to be something per-process or something like that, don't know.